pkgs/development/compilers/blueprint/default.nix

```diff
@@ -0,0 +1,50 @@
+{ python3
+, stdenv
+, fetchFromGitLab
+, gobject-introspection
+, lib
+, meson
+, ninja
+}:
+stdenv.mkDerivation rec {
+  pname = "blueprint-compiler";
+  version = "0.pre+date=2022-05-17";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "jwestman";
+    repo = pname;
+    rev = "06278a64248cec92bb95a958eadfba453943c061";
+    sha256 = "sha256-ukvqWvYl4NohKEm6W50eRZ9fmKD4vhH0yzgrCf5HVts=";
+  };
+
+  # Requires pythonfuzz, which I've found difficult to package
+  doCheck = false;
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    python3.pkgs.wrapPython
+  ];
+  buildInputs = [
+    gobject-introspection
+    python3
+  ];
+
+  preFixup = ''
+    makeWrapperArgs+=(
+      --prefix PATH : "${lib.makeBinPath [ gobject-introspection ]}"
+    )
```

Comment:
There should be a hook which does that already.

Reply:
`wrapPythonPrograms` does that but it would require propagating them.